src/main/java/com/google/edith/interfaces/ReceiptFileHandlerInterface.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith.interfaces;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.FileInfo;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Operations for handling blobs. */
+public interface ReceiptFileHandlerInterface {
+  void serveBlob(HttpServletResponse response, List<FileInfo> fileKeys) throws IOException;
+
+  List<FileInfo> getUploadedFileUrl(HttpServletRequest request, String formInputElementName);
+
+  BlobKey getBlobKey(List<FileInfo> fileKeys);
```

Comment:
Typically we leave a javadoc on the interface methods and just add an override annotation on the implementing class. If the implementing class has details specific to the impl, then we can include in that javadoc. It's ok to not have a javadoc on the impl class

Reply:
Thanks for the suggestion.
I moved the javadocs to interface method and just used the @Override annotation on the implementing class.